File: src/main/java/io/github/dylmeadows/eontimer/model/config/ActionMode.java
package io.github.dylmeadows.eontimer.model.config;

import io.github.dylmeadows.common.javafx.util.Choice;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ActionMode implements Choice {
    AUDIO("Audio"),
    VISUAL("Visual"),
    AV("A/V"),
    NONE("None");

    private final String text;
}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/settings/ActionMode.kt
package io.github.dylmeadows.eontimer.model.settings

import io.github.dylmeadows.common.javafx.util.Choice

enum class ActionMode(private val text: String) : Choice {
    AUDIO("Audio"),
    VISUAL("Visual"),
    AV("A/V"),
    NONE("None");

    override fun getText(): String {
        return text
    }
}